app/component/summary/OriginDestinationBar.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { swapEndpoints } from '../../action/EndpointActions';
+
+import { intlShape } from 'react-intl';
+
+import Icon from '../icon/icon';
+import OneTabSearchModal from '../search/one-tab-search-modal';
+
+class OriginDestinationBar extends React.Component {
+
+  static contextTypes = {
+    getStore: React.PropTypes.func.isRequired,
+    executeAction: React.PropTypes.func.isRequired,
+    intl: intlShape.isRequired,
+  };
+
+  constructor() {
+    super();
+
+    this.state = {
+      origin: undefined,
+      destination: undefined,
+      tabOpen: false,
+    };
+
+    this.componentWillMount = this.componentWillMount.bind(this);
+    this.componentWillUnmount = this.componentWillUnmount.bind(this);
+    this.onEndpointChange = this.onEndpointChange.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+    this.openSearch = this.openSearch.bind(this);
+  }
+
+  componentWillMount() {
+    this.onEndpointChange();
+    this.context.getStore('EndpointStore').addChangeListener(this.onEndpointChange);
+  }
+
+  componentWillUnmount() {
+    this.context.getStore('EndpointStore').removeChangeListener(this.onEndpointChange);
+  }
+
+  onEndpointChange() {
+    this.setState({
+      origin: this.context.getStore('EndpointStore').getOrigin(),
+      destination: this.context.getStore('EndpointStore').getDestination(),
+    });
+  }
+
+  closeModal() {
+    this.setState({
+      tabOpen: false,
+    });
+  }
+
+  openSearch(tab) {
+    this.setState({
+      tabOpen: tab,
+    });
+  }
+
+  render() {
+    const ownPosition = this.context.intl.formatMessage({
+      id: 'own-position',
+      defaultMessage: 'Your current location',
+    });
+
+    let initialValue = '';
+
+    if (this.state[this.state.tabOpen]) {
+      initialValue = this.state[this.state.tabOpen].useCurrentPosition ?
```

Comment:
Perhaps use `if` here, as you are assigning the result to a non-const. Or I would at least have the both sides of the colon : on the same line.